src/net/ipv4.c

```diff
@@ -890,9 +912,52 @@ static int ipv4_settings ( int ( * apply ) ( struct net_device *netdev,
 		/* Get default gateway, if present */
 		fetch_ipv4_setting ( settings, &gateway_setting, &gateway );
 
-		/* Apply settings */
-		if ( ( rc = apply ( netdev, address, netmask, gateway ) ) != 0 )
-			return rc;
+		/* Configure route (-s) */
+		uint8_t * raw_data = netdev->static_routes;
+		uint8_t option_len = netdev->static_routes_len;
+
+
+		/* Check if static routes available
+		 * The minimum length for DHCP option 121 is 5 bytes */
+		if ( option_len >= 5 ) {
+			uint8_t i = 0;
+			uint8_t j;
+			uint8_t addr_octets_len;
+			uint8_t prefix_len;
+
+			union ipv4_conversion address_conversion = {0};
+			union ipv4_conversion gateway_conversion = {0};
+
+			/* Extract routes and add them to routing table */
+			while ( i < option_len ) {
+				prefix_len = raw_data[i];
+				i++;
+
+				/* Netmask */
+				addr_octets_len = prefix_len / 8;
+				netmask.s_addr = prefix_len_to_mask ( prefix_len );
+
+				/* Address */
+				for ( j = 0; j < addr_octets_len; j++ )
```

Comment:
This won't work if a default route is handed out in any position but the first within the option. Allow me to explain:

Let's say the dhcp server hands out a route `172.31.254.0/24 gw 192.168.178.243` followed by a default route (5 bytes, `0` plus the gateway).  
After processing the first 8 bytes of the option (i.e. `18 ac 1f fe c0 a8 b2 f3` which is the route `172.31.254.0/24 gw 192.168.178.243`), `address_conversion.octets` will contain data. But the 9th byte is going to be 0 (the network/mask part of the default route) so `prefix_len` is 0, so `addr_octets_len` is 0 and we skip the for loop, never chaning the data in `address_conversion.octets`.

So instead of `address` being `0.0.0.0`, it's going to be the stale value from the route processed before (`172.31.254.0` in this example)